File: workshop-exercises/02-clean-code/go/application/interfaces.go
package application

type OrderServiceInterface interface {
	ProcessOrder(order OrderData) (string, error)
}

type PaymentProcessorInterface interface {
	ProcessPayment(amount float64) (string, error)
}

type DiscountServiceInterface interface {
	CalculateDiscount(amount float64, customerType string) (float64, error)
}

type OrderData struct {
	Amount       float64
	Customer     string
	CustomerType string
}

File: workshop-exercises/02-clean-code/csharp/Application/Interfaces/Interfaces.cs
using Workshop.Application.Models;

namespace Workshop.Application.Interfaces
{
    public interface IOrderService
    {
        string ProcessOrder(OrderData order);
    }

    public interface IPaymentProcessor
    {
        string ProcessPayment(double amount);
    }

    public interface IDiscountService
    {
        double CalculateDiscount(double amount, string customerType);
    }
}